NBitcoin.Tests/NetworkAddressTests.cs

```diff
@@ -0,0 +1,560 @@
+using System.IO;
+using System.Linq;
+using System.Threading;
+using NBitcoin;
+using Xunit;
+using System;
+using NBitcoin.Protocol;
+using System.Net;
+using NBitcoin.DataEncoders;
+
+namespace NBitcoin.Tests
+{
+	public class NetworkAddressTests
+	{
+		[Fact]
+		[Trait("UnitTest", "UnitTest")]
+		public void CanParseSpecialAddresses()
+		{
+			var addr = new NetworkAddress();
+
+			// TORv2
+			Assert.True(addr.SetSpecial("6hzph5hv6337r6p2.onion"));
+			Assert.True(addr.IsTor);
+
+			Assert.True(addr.IsAddrV1Compatible);
+			Assert.Equal("6hzph5hv6337r6p2.onion", addr.ToAddressString());
+
+			// TORv3
+			var torv3_addr = "pg6mmjiyjmcrsslvykfwnntlaru7p5svn6y2ymmju6nubxndf4pscryd.onion";
+			Assert.True(addr.SetSpecial(torv3_addr));
+			Assert.True(addr.IsTor);
+
+			Assert.False(addr.IsAddrV1Compatible);
+			Assert.Equal(addr.ToAddressString(), torv3_addr);
+
+			// TORv3, broken, with wrong checksum
+			Assert.False(addr.SetSpecial("pg6mmjiyjmcrsslvykfwnntlaru7p5svn6y2ymmju6nubxndf4pscsad.onion"));
+
+			// TORv3, broken, with wrong version
```

Comment:
Do you mean that Tor v3 addresses must end with `d`?

Reply:
source?

Reply:
https://old.reddit.com/r/onions/wiki/index#wiki_types_of_.onion_addresses